privaterelay/utils.py

```diff
@@ -27,35 +27,48 @@ def get_countries_info_from_request_and_mapping(request, mapping):
     }
 
 
-def _get_cc_from_request(request):
+def _get_cc_from_request(request, premium_mapping):
     if "X-Client-Region" in request.headers:
         return request.headers["X-Client-Region"].lower()
     if "Accept-Language" in request.headers:
-        return get_premium_country_lang(request.headers["Accept-Language"])[0]
+        return get_premium_country_lang(
+            request.headers["Accept-Language"], premium_mapping
+        )[0]
     if settings.DEBUG:
         return "us"
     return "us"
 
 
-def get_premium_country_lang(accept_lang, cc=None):
+def get_premium_country_lang(accept_lang, mapping, cc=None):
     lang = accept_lang.split(",")[0]
     lang_parts = lang.split("-") if lang and "-" in lang else [lang]
     lang = lang_parts[0].lower()
     if cc is None:
-        cc = lang_parts[1] if len(lang_parts) == 2 else lang_parts[0]
+        cc = lang_parts[1] if len(lang_parts) >= 2 else lang_parts[0]
```

Comment:
My main reason for this change is to match the logic in `fxa_locale_in_premium_country`, which uses `>= 2`:

https://github.com/mozilla/fx-private-relay/blob/cbbc3b5813799aff6b1d3573e35ba5aa29604c46/emails/models.py#L175-L180

For the headers we'd expect to see, none have more than 2 parts, and the country code is the second part. Here's the ones from Firefox default headers:

```
 "az-AZ, az, en-US, en"
"bn-in, bn, en-us, en"
"bo-CN,bo-IN,bo,en-US, en"
"bs-ba, bs, en-us, en"
"ca-valencia, ca, en-us, en"
"cy-GB, cy, en-US, en"
"el-GR, el, en-US, en"
"en-CA, en-US, en"
"en-GB, en"
"es-AR, es, en-US, en"
"es-CL, es, en-US, en"
"es-ES, es, en-US, en"
"es-MX, es, en-US, en"
"fa-ir, fa, en-us, en"
"fi-fi, fi, en-us, en"
"fur-IT, fur, it-IT, it, en-US, en"
"fy-nl, fy, nl, en-us, en "
"ga-ie, ga, en-ie, en-gb, en-us, en"
"gd-gb, gd, en-gb, en-us, en"
"gl-gl, gl, en-us, en"
"gu-in, gu, en-us, en"
"hi-in, hi, en-us, en"
"hu-hu, hu, en-US, en"
"hy-AM,hy,en-us,en"
"ilo-PH, ilo, en-US, en"
"it-IT, it, en-US, en"
"ka-GE,ka,en-US,en"
"kab-DZ,kab,fr-FR,fr,en-US,en"
"kn-IN, kn, en-us, en"
"ko-KR, ko, en-US, en"
"mk-mk, mk, en-us, en"
"ml-in, ml, en-US, en"
"mr-IN, mr, en-US, en"
"nb-no, nb, no-no, no, nn-no, nn, en-us, en"
"ne-NP, ne, en-US, en"
"nn-no, nn, no-no, no, nb-no, nb, en-us, en"
"pt-BR, pt, en-US, en"
"pt-PT, pt, en, en-US"
"ro-RO, ro, en-US, en-GB, en"
"ru-RU, ru, en-US, en"
"si-LK, si, en-US, en"
"sr-RS, sr, en-US, en"
"sv-SE, sv, en-US, en"
"ta-IN, ta, en-US, en"
"te-in, te, en-us, en"
"tl-PH, tl, en-US, en"
"tr-TR, tr, en-US, en"
"tr-TR, tr, en-US, en"
"uk-UA, uk, en-US, en"
"ur-PK, ur, en-US, en"
"vi-vn, vi, en-us, en"
"xh-ZA, xh, en-US, en"
"zh-CN, zh, zh-TW, zh-HK, en-US, en"
"zh-tw, zh, en-us, en"
```

The three-element ones are not as common, and would probably be there because a user changed their header. Here are some examples from [IANA's list](https://www.iana.org/assignments/language-tags/language-tags.xhtml):

* `de-AT-1901` - German, Austrian variant, traditional orthography
* `de-CH-1996` - German, Swiss variant, orthography of 1996
* `de-DE-1901` - German, German variant, traditional orthography
* `en-GB-oed` - English, Oxford English Dictionary spelling
* `sgn-CH-de` - Swiss German Sign Language (featured in the unit test)
* `zh-Hans-CN` - PRC Mainland Chinese in simplified script
* `zh-Hans-HK` - Hong Kong Chinese in simplified script
* `zh-Hans-TW` - Taiwan Chinese in simplified script

As you can see, sometimes the middle bit is a region, and sometimes it is not.

I don't like this code that mixes languages and regions. However, it probably works 99% of the time, and I feel like fixing it with more exact checks is a different PR.